examples/mnemonic_to_descriptors.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2020-2021 Bitcoin Dev Kit Developers
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+use bdk::bitcoin::secp256k1::Secp256k1;
+use bdk::bitcoin::util::bip32::DerivationPath;
+use bdk::bitcoin::Network;
+use bdk::keys::bip39::{Language, Mnemonic, WordCount};
+use bdk::keys::DescriptorKey::Secret;
+use bdk::keys::KeyError::Message;
+use bdk::keys::{
+    DerivableKey, DescriptorKey, ExtendedKey, GeneratableKey, GeneratedKey, IntoDescriptorKey,
+};
+use bdk::miniscript::Segwitv0;
+use bdk::Error as BDK_Error;
+use std::error::Error;
+use std::str::FromStr;
+
+/// This example demonstrates how to generate a mnemonic phrase
+/// using BDK and use that to generate a master extended private key.
+/// It then uses the master extended private key with a derivation
+/// path to generate child extended private and public keys. Finally,
+/// it uses the derivation path and these keys to get the descriptors.
+fn main() -> Result<(), Box<dyn Error>> {
+    let secp = Secp256k1::new();
+
+    // In this example we are generating a 12 words mnemonic phrase
+    // but it is also possible generate 15, 18, 21 and 24 words
+    // using their respective `WordCount` variant.
+    let mnemonic: GeneratedKey<_, Segwitv0> =
+        Mnemonic::generate((WordCount::Words12, Language::English))
+            .map_err(|_| BDK_Error::Generic("Mnemonic generation error".to_string()))?;
+
+    let mnemonic = mnemonic.into_key();
+
+    // We don't want a passphrase on the mnemonic so we use None
+    let xkey: ExtendedKey = (mnemonic.clone(), None).into_extended_key()?;
+    let master_xprv = xkey.into_xprv(Network::Testnet).ok_or_else(|| {
+        BDK_Error::Generic("Privatekey info not found (should not happen)".to_string())
+    })?;
+
+    let fingerprint = master_xprv.fingerprint(&secp);
+    let phrase = mnemonic.to_string();
+
+    println!("Mnemonic phrase: {}", phrase);
+    println!("Xprv fingerprint: {}", fingerprint);
+
+    // You can replace this derivation path with one of your choosing
+    let path = DerivationPath::from_str("m/84h/0h/0h").unwrap();
+
+    // The second item in the mnemonic tuple represents the passphrase.
+    let derived_xprv_desc_key: DescriptorKey<Segwitv0> =
+        ((mnemonic, None), path).into_descriptor_key()?;
+
+    if let Secret(desc_seckey, _, _) = derived_xprv_desc_key {
+        let desc_pubkey = desc_seckey
+            .as_public(&secp)
+            .map_err(|e| BDK_Error::Generic(e.to_string()))?;
+        println!("xpub {}", desc_pubkey);
+        println!("xprv {}", desc_seckey);
+    } else {
+        return Err(Box::new(BDK_Error::Key(Message(
+            "Invalid key variant".to_string(),
+        ))));
+    }
```

Comment:
As per the description of the example, it would be good to extend this a little further and create a full descriptor string with the `Xpriv`.. And then another step of showing how to create a watch only descriptor with the `Xpub` part alone..

Reply:
I will do that!